src/nupic/utils/VectorHelpers.hpp

```diff
@@ -109,6 +109,38 @@ class VectorHelpers
     }
     return activeColumns;
   }
+  /**
+   * A sparse array version of cellsToColumns( ).
+   */
+  static std::vector<UInt> sparse_cellsToColumns(const std::vector<UInt>& cellsSparse, 
+                                                 const UInt cellsPerColumn)
+  {
+    std::vector<UInt> activeColumns;
+
+    // loop over the whole (active) cells array
+    // saving the column indexes.
+    UInt prev = std::numeric_limits<UInt>::max();
+    for(auto cellIdx: cellsSparse) {
+      UInt colIdx = cellIdx / cellsPerColumn;
+      if (colIdx != prev) activeColumns.push_back(colIdx);
+      prev = colIdx;
+    }
+    return activeColumns;
+  }
+
+  /**
+   * Create a Union of two vectors (An OR of the two).
+   * The values are assumed to be sorted, sparse indexes.
+   */
+  static void unionOfVectors(std::vector<UInt>& out, 
```

Comment:
There is not a method to make SDR Unions but I can make one...

It should be relatively simple since we do have an SDR-Intersection class which we can copy from.

Reply:
>  But we'll need test

Oh, good point.  I need to add some tests for those two new functions.